src/digest.rs

```diff
@@ -475,17 +466,30 @@ pub const MAX_OUTPUT_LEN: usize = 512 / 8;
 pub const MAX_CHAINING_LEN: usize = MAX_OUTPUT_LEN;
 
 fn sha256_format_output(input: State) -> Output {
-    let input = unsafe { &input.as32 };
-    Output {
-        as32: input.map(BigEndian::from),
-    }
+    let input = unsafe { input.as32 };
+    format_output::<_, _, { core::mem::size_of::<u32>() }>(input, u32::to_be_bytes)
 }
 
 fn sha512_format_output(input: State) -> Output {
-    let input = unsafe { &input.as64 };
-    Output {
-        as64: input.map(BigEndian::from),
-    }
+    let input = unsafe { input.as64 };
+    format_output::<_, _, { core::mem::size_of::<u64>() }>(input, u64::to_be_bytes)
+}
+
+#[inline]
```

Comment:
I could be wrong about this, but my understanding is that inside of a crate (as opposed to between crates), `#[inline]` has little to no effect.